Fix #466. Fix variable mixup in MimeType.isCompatible

dvntucker opened the pull request:
Fixed a mistaken variable in isCompatible that was causing false positives. XML error MimeType was  reporting as compatible with image mime types, which caused GWC to mistakenly save service exceptions as images.

smithkm commented on geowebcache/core/src/main/java/org/geowebcache/mime/MimeType.java:
```diff
@@ -198,7 +198,8 @@ public int hashCode() {
 
 
     public boolean isCompatible(String otherMimeType) {
```
Could you update the javadoc to clarify how this is used.  Something like "Checks if the given MIME type is a subtype of this one."

smithkm commented on geowebcache/core/src/main/java/org/geowebcache/mime/MimeType.java:
```diff
@@ -198,7 +198,8 @@ public int hashCode() {
     public boolean isCompatible(String otherMimeType) {
-        return mimeType.equalsIgnoreCase(otherMimeType) || otherMimeType.startsWith(otherMimeType);
+        return mimeType.equalsIgnoreCase(otherMimeType) ||
+            otherMimeType.startsWith(mimeType);
```
It looks like this makes it case insensitive if it's otherwise an exact match, but case sensitive if matching a subtype, which is really an odd thing to do.

dvntucker commented:
Updated to have JavaDoc, and do a case-insensitive prefix check (also a null check before that to prevent NPE).

I also changed the test case to use JPEG, since I just noticed that PNG actually already has its own implementation of isCompatible which... maybe isn't needed any longer?
